acis_thermal_check/state_builder.py

```diff
@@ -0,0 +1,294 @@
+import os
+import Ska.DBI
+from Chandra.Time import DateTime
+from pprint import pformat
+import Chandra.cmd_states as cmd_states
+import numpy as np
+from acis_thermal_check.utils import globfile
+
+class StateBuilder(object):
+    """
+    This is the base class for all StateBuilder objects. It
+    should not be used by itself, but subclassed.
+
+    Parameters
+    ----------
+    thermal_check : :class:`~acis_thermal_check.main.ACISThermalCheck` object
+        The ACISThermalCheck object this StateBuilder is attached to.
+    """
+    def __init__(self, thermal_check):
+        self.thermal_check = thermal_check
+        self.logger = self.thermal_check.logger
+
+    def set_options(self, args):
+        """
+        Give the StateBuilder arguments that were passed in
+        from the command line
+
+        Subclasses may want to overload this to do other things
+        with the arguments. 
+
+        Parameters
+        ----------
+        args : ArgumentParser object
+            Command line arguments
+        """
+        self.args = args
+
+    def get_predict_states(self, tlm):
+        """
+        Get the states used for the thermal prediction.
+
+        Parameters
+        ----------
+        tlm : dictionary
+            Dictionary containg temperature and other telemetry
+        """
+        pass
```

Comment:
Maybe raise `NotImplementedError` instead?

Reply:
Yes, I'll raise an error here and in the other method.